app/javascript/components/rooms/room/join/RoomJoin.jsx

```diff
@@ -38,10 +38,16 @@ export default function RoomJoin() {
   const path = encodeURIComponent(location.pathname);
 
   useEffect(() => { // set cookie to return to if needed
-    document.cookie = `location=${path};path=/`;
-  });
+    document.cookie = `location=${path};path=/;`;
+
+    return () => { // delete redirect location when unmounting
+      document.cookie = 'location=;path=/;expires=Thu, 01 Jan 1970 00:00:00 GMT';
+    };
+  }, []);
 
   const handleJoin = (data) => {
+    document.cookie = 'location=;path=/;expires=Thu, 01 Jan 1970 00:00:00 GMT'; // delete redirect location
```

Comment:
Joining for some reason does not unmount the component, so adding an addition check here